text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,125 @@
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md
+
+# Summary
+[summary]: #summary
+
+This RFC describes the implementation the following commands:
+* `pack publish-buildpack`
+* `pack yank-buildpack`
+
+# Motivation
+[motivation]: #motivation
+
+The [Client-Side Registry RFC (0022)](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md) established that two CLI commands would allow users to add/remove buildpacks in the registry, but the implementation of those commands was vauge. In this document we describe the details of how they will work.
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Adding a Buildpack to the Registry
+
+When a buildpack author would like to publish a version of a buildpack on the registry, they will use the `pack publish-buildpack <url>` command. This command will create a Github Issue on the `https://github.com/buildpack/registry` (or similarly named) Github repo requesting the addition the new buildpack version to the index.
+
+## Yanking a Buildpack from the Registry
+
+Sometimes a buildpack author may have pushed up a bad version that they wish to not be available in the index. In order to not break builds, it will not be possible to fully remove an entry from the index. Instead, the entry in the index will be marked as "yanked". This information can then be used when resolving which buildpacks to fetch.
+
+The user will run the `pack yank-buildpack <namespace>/<name>@<version>` command to yank a buildpack. If a buildpack author wants to undo the yank and make the buildpack version available in the index, they can run the same command and use the `--undo` flag.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## `pack publish-buildpack <url>`
+
+Arguments:
+
+* `url` - the location of a buildpackage _in an Docker registry_ to be released in the buildpack registry
+
+Steps:
+
+1. Pulls the image manifest for the buildpackage at `url` if it does not already exist locally.
+1. Reads the following information from the image manifest (if any of these are missing, the command fails):
+    - `id` (must include a `namespace` and `name` in the form `<namespace>/<name>`)
+    - `version`
+1. Opens a link to a new Github Issue in the user's browser
+    - The user must be logged into Github
+    - The link will use [Github query parameters](https://help.github.com/en/github/managing-your-work-on-github/about-automation-for-issues-and-pull-requests-with-query-parameters) to pre-populate the title, body, and labels.
+    - The issue will be opened against the `https://github.com/buildpacks/registry` repo, which is *not* the index. The `buildpacks/registry` defines the [namespace ownership as described in RFC-0022](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md#namespace-ownership).
```

Comment:
Yea, I'm starting to think the same. In part because if `publish-buildpack` doesn't work against the registry-index, there's this weird asymmetry between `build -R` and `publish-buildpack` (talking to two different places).

This would also mean that if the default registry or `-R` value is a Github URL, we could still perform these steps (allowing people to more easily fork the index).